qemu/tests/qemu_guest_agent.py

```diff
@@ -839,6 +818,27 @@ def gagent_check_reboot_shutdown(self, test, params, env):
             except Exception:
                 pass
 
+    @error_context.context_aware
+    def _action_before_fsfreeze(self, *args):
+        session = self._get_session(self.params, None)
+        self._open_session_list.append(session)
+
+    @error_context.context_aware
+    def _action_after_fsfreeze(self, *args):
+        error_context.context("Verfiy FS is frozen in guest.", logging.info)
+
+        if not self._open_session_list:
+            self.test.error("Could not find any opened session")
+        # Use the last opened session to send cmd.
+        session = self._open_session_list[-1]
+
+        try:
+            session.cmd(self.params["gagent_fs_test_cmd"])
+        except aexpect.ShellTimeoutError:
+            logging.info("FS is frozen as expected,can't write in guest.")
+        else:
+            self.test.fail("FS is not frozen,still can write in guest.")
+
```

Comment:
Why move this section without any changes?

Reply:
As  `def gagent_check_reboot_shutdown() function`(merged recently) separate fsfreeze/fsthaw function.